client/components/forms/form-text-input/index.jsx

```diff
@@ -30,7 +30,10 @@ export default class FormTextInput extends PureComponent {
 	}
 
 	render() {
-		const props = omit( this.props, 'isError', 'isValid', 'selectOnFocus' );
+		const {
+			inputRef,
+			...props
```

Comment:
Spread destructure and `omit` are each `O(n)` to object properties and serve a similar purpose. Would be nice if we could limit it to just one or the other, e.g.

```js
// eslint-disable-next-line no-unused-vars
const { inputRef, isError, isValid, selectOnFocus, ...props } = this.props;
```

or:

```js
const props = omit( this.props, 'isError', 'isValid', 'selectOnFocus', 'inputRef' );
const { inputRef } = this.props;
```

Not a huge optimization, especially since it's a pure component anyways, but maybe worth considering for a common component.

See: http://babeljs.io/repl/#?babili=false&evaluate=false&lineWrap=false&presets=es2015%2Cstage-2&targets=&browsers=ie%2011&builtIns=true&debug=false&code=const%20%7B%0A%20%20inputRef%2C%0A%20%20...props%0A%7D%20%3D%20omit(%20%7B%7D%2C%20'isError'%2C%20'isValid'%2C%20'selectOnFocus'%20)%3B&plugins=transform-runtime

See: https://github.com/eslint/eslint/issues/4880